src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,17 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Machine> {
+    @Override
+    public List<Machine> get() {
```

Comment:
The method `get()` should return a list of `Truck` instances instead of `Machine`. This ensures type safety and prevents the need for casting when using the list.